src/Components/Components/src/Forms/InputComponents/InputCheckbox.cs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Components.RenderTree;
+
+namespace Microsoft.AspNetCore.Components.Forms
+{
+    /* This is exactly equivalent to a .razor file containing:
+     *
+     *    @inherits InputBase<bool>
+     *    <input type="checkbox" bind="@CurrentValue" class="@CssClass" />
+     *
+     * The only reason it's not implemented as a .razor file is that we don't presently have the ability to compile those
+     * files within this project. Developers building their own input components should use Razor syntax.
+     */
```

Comment:
Do you know the reason this doesn't work? Or we just haven't tried yet.

Reply:
We have to change the `Sdk` etc. AFAIK it can be done, but I am trying to keep this change scoped.